stetho/src/main/java/com/facebook/stetho/inspector/protocol/module/DOM.java

```diff
@@ -2,30 +2,315 @@
+    }
+
+    @Override
+    public void onAttributeRemoved(Object element, String name) {
+      AttributeRemovedEvent message = new AttributeRemovedEvent();
+      message.nodeId = getOrCreateNodeIdForElement(element);
+      message.name = name;
+      mPeerManager.sendNotificationToPeers("DOM.attributeRemoved", message);
+    }
+
+    @Override
+    public void onChildInserted(Object parentElement, Object previousElement, Object childElement) {
+      ChildNodeInsertedEvent message = new ChildNodeInsertedEvent();
+      message.parentNodeId = getOrCreateNodeIdForElement(parentElement);
+
+      message.previousNodeId = (previousElement == null)
+          ? -1  // using -1 was just a guess, and it seemed to work. this is for the case where we
+                // go from 0 to 1 children, or if we just happen to be inserting at index 0
+          : getOrCreateNodeIdForElement(previousElement);
+
+      message.node = createNodeForElement(childElement);
+      mPeerManager.sendNotificationToPeers("DOM.childNodeInserted", message);
+    }
+
+    @Override
+    public void onChildRemoved(Object parentElement, Object childElement) {
+      ChildNodeRemovedEvent message = new ChildNodeRemovedEvent();
+      message.parentNodeId = getOrCreateNodeIdForElement(parentElement);
+      message.nodeId = getOrCreateNodeIdForElement(childElement);
+      mPeerManager.sendNotificationToPeers("DOM.childNodeRemoved", message);
+      unhookElement(childElement);
+    }
+  }
+
+  private static class AttributeModifiedEvent {
+    @JsonProperty(required = true)
+    public int nodeId;
+
+    @JsonProperty(required = true)
+    public String name;
+
+    @JsonProperty(required = true)
+    public String value;
+  }
+
+  private static class AttributeRemovedEvent {
+    @JsonProperty(required = true)
+    public int nodeId;
+
+    @JsonProperty(required = true)
+    public String name;
+  }
+
+  private static class ChildNodeInsertedEvent {
+    @JsonProperty(required = true)
+    public int parentNodeId;
+
+    @JsonProperty(required = true)
+    public int previousNodeId;
+
+    @JsonProperty(required = true)
+    public Node node;
+  }
+
+  private static class ChildNodeRemovedEvent {
+    @JsonProperty(required = true)
+    public int parentNodeId;
+
+    @JsonProperty(required = true)
+    public int nodeId;
+  }
+
+  @ChromeDevtoolsMethod
+  public void enable(JsonRpcPeer peer, JSONObject params) {
+    mPeerManager.addPeer(peer);
+  }
+
+  @ChromeDevtoolsMethod
+  public void disable(JsonRpcPeer peer, JSONObject params) {
+    mPeerManager.removePeer(peer);
   }
 
   @ChromeDevtoolsMethod
   public JsonRpcResult getDocument(JsonRpcPeer peer, JSONObject params) {
-    Node node = new Node();
-    node.nodeId = 1;
-    node.nodeType = 1;
-    node.nodeName = "";
-    node.nodeValue = "";
-    return node;
+    Object rootElement = mDOMProvider.getRootElement();
+    if (rootElement == null) {
```

Comment:
You can't throw because this is a reasonable case.  Start the sample app, hit back to dismiss all UI, open chrome://inspect.  This case should occur.  But what actually happens in the DevTools UI?